### DISCUSSION TITLE
Implement get_result & get_results on mysql, sqlite backend by using `SELECT last_insert_rowid()` and `SELECT LAST_INSERT_ID();`

### DISCUSSION THREAD

**Discussion by XPH0816** [diesel-rs/diesel#4045] [Ideas]
# Feature Review

>  Since MySQL and SQLite no support get_result & get_results, Here's some idea to implement it.

## Description for `SELECT last_insert_rowid();` 
 > Only for **SQLite** 

### Single Insert
The `last_insert_rowid()` return the last insert row id on the auto increment id 

### Batch Insert
Same as Single Insert 

## Description for `SELECT last_insert_id();` 
> Only for **MySQL** 

### Single Insert
The `last_insert_id()` return the last insert row id on the auto increment id 

### Batch Insert
The `last_insert_id()` return the first insert rows id on the auto increment id

**Comment by weiznich** [comment]
I'm sorry, but its really not clear how you propose to implement these methods exactly. I see no way how `last_insert_rowid()` could be used in the general case there as there are no guarantees that there is even an autoincremented id nor that it might return the right results in a highly concurrent scenario.

**Reply by XPH0816** [reply]
I'm sorry too for my poor writing skill, but I think is suitable for RETURNING Clause on SQLite3 that the version is >= 3.35.0.
I hope this can be helped on this project 🤔.

References Link:
https://www.sqlite.org/lang_returning.html

**Reply by weiznich** [reply]
That's already supported behind the corresponding feature flag.

**Reply by XPH0816** [reply]
Ohhh, Thanks for the patient that U reply me.
Now I mark this discussion as done.
Sorry if got inconvenience caused.
